src/frontend/src/handler/query.rs

```diff
@@ -218,18 +227,23 @@ pub async fn handle_query(
         }
 
         // update some metrics
-        if query_mode == QueryMode::Local {
-            session
-                .env()
-                .frontend_metrics
-                .latency_local_execution
-                .observe(query_start_time.elapsed().as_secs_f64());
+        match query_mode {
+            QueryMode::Local => {
+                session
+                    .env()
+                    .frontend_metrics
+                    .latency_local_execution
+                    .observe(query_start_time.elapsed().as_secs_f64());
 
-            session
-                .env()
-                .frontend_metrics
-                .query_counter_local_execution
-                .inc();
+                session
+                    .env()
+                    .frontend_metrics
+                    .query_counter_local_execution
+                    .inc();
+            }
+            QueryMode::Distributed => {
+                session.env().record_complete_a_distributed_query();
```

Comment:
Can we move this metrics into `QueryManager`? In general, I want to add following metrics:
1. distributed_running_query: (Gauge, number of running queries)
2. distributed_query_counter: (Counter, total number of distributed query)
3. rejected_distributed_query_counter: (Counter, total number of rejected distributed query)
4. distributed_query_latency: (Histogram, query execution time of distributed query)

Also we don't need to handle these in callback, distributed query has clear lifetime definition and event processing logic, so we just need to handle is in QueryManager.